projects/igniteui-angular/src/lib/select/select-positioning-strategy.ts

```diff
@@ -31,201 +25,231 @@ export class SelectPositioningStrategy extends ConnectedPositioningStrategy impl
-        contentElement.style.top = `${CURRENT_POSITION_Y - this.itemTextToInputTextDiff}px`;
-        this.deltaY = CURRENT_POSITION_Y -
-            (this.select.input.nativeElement.getBoundingClientRect() as DOMRect).top - this.itemTextToInputTextDiff;
-    }
-
-    private positionAndScrollTop(contentElement: HTMLElement, outBoundsAmount: number) {
-        contentElement.style.top = `${this.viewPort.top + this.defaultWindowToListOffset}px`;
-        contentElement.firstElementChild.scrollTop += outBoundsAmount + this.itemTextToInputTextDiff + this.defaultWindowToListOffset;
-        this.deltaY = this.viewPort.top + this.defaultWindowToListOffset -
-            (this.select.input.nativeElement.getBoundingClientRect() as DOMRect).top;
-    }
-
-    private getItemsOutOfView(contentElement: HTMLElement, itemHeight: number): {
-        'currentScroll': number,
-        'remainingScroll': number
-    } {
-        if (contentElement.firstElementChild.scrollHeight <= contentElement.firstElementChild.clientHeight) {
-            return {
-                'currentScroll': 0,
-                'remainingScroll': 0
-            };
-        }
-        const currentScroll = contentElement.firstElementChild.scrollTop;
-        const remainingScroll = this.select.items.length * itemHeight - currentScroll - this.listContainerBoundRect.height;
-        return {
-            'currentScroll': currentScroll,
-            'remainingScroll': remainingScroll
+    // Global variables required for cases of !initialCall (page scroll/overlay repositionAll)
+    private global_yOffset = 0;
+    private global_xOffset = 0;
+    private global_styles: SelectStyles = {};
+
+    position(contentElement: HTMLElement, size: Size, document?: Document, initialCall?: boolean): void {
+        // 1st: Check if the interaction item is visible. If NO it should be scrolled in view.
+        //      Adjust the new container position with the scrolled amount.
+        // 2nd: Try position the container after the item is scrolled.
+        // 3rd: If unsuccessful, position the overlay container on TOP/ABOVE or BOTTOM/BELLOW of the input. BOTTOM/BELLOW is preferred.
+        // 4th: On page scroll, persist the ddl container position relative to its target element.
+
+        const rects = super.calculateElementRectangles(contentElement);
+        // selectFit obj, to be used for both cases of initialCall and !initialCall(page scroll/overlay repositionAll)
+        const selectFit: SelectFit = {
+            verticalOffset: this.global_yOffset,
+            horizontalOffset: this.global_xOffset,
+            targetRect: rects.targetRect,
+            contentElementRect: rects.elementRect,
+            styles: this.global_styles
         };
+
+        if (initialCall) {
+            selectFit.viewPortRect = Util.getViewportRect(document);
+
+            // Fill in the required selectFit object properties.
+            this.calculateVariables(selectFit);
```

Comment:
At the end `calculateVariables` will remain two lines of code. I would remove this method like this:

```
            selectFit.viewPortRect = Util.getViewportRect(document);
            selectFit.itemElement = this.getInteractionItemElement();
            selectFit.itemRect = selectFit.itemElement.getBoundingClientRect();

            // Calculate input and selected item elements style related variables
            selectFit.styles = this.calculateStyles(selectFit);

```

And do not call this twice - `selectFit.viewPortRect = Util.getViewportRect(document);` :)

Reply:
done